### ISSUE TITLE
Cone Phantom

### ISSUE THREAD

**Issue by tknopp** [JuliaImageRecon/ImagePhantoms.jl#57]
Within magnetic particle imaging we are using cone phantoms:

<img width="618" alt="image" src="https://user-images.githubusercontent.com/3109172/196029508-56307e29-5735-4d31-928c-ed8ca3cb0326.png">

Here is some hacked code that generates a synthetic cone

```
function makeConePhantom(N, Ry=N[2]÷4, Rz=N[3]÷3, L=N[1]÷2)
    I = zeros(Float32, N...)
    for x=(N[1]÷2-L÷2):(N[1]÷2+L÷2-1)  
       h = (L-x+N[1]÷2-L÷2)/L
       I[x, :, :] = [sqrt((y-N[2]÷2)^2/Ry^2 + (z-N[3]÷2)^2/Rz^2) <= h for y=1:N[2], z=1:N[3]]
    end
    return I
  end
```

Would something like this fit into `ImagePhantoms.jl`? Of course a proper version derived from triangle.

**Comment by JeffFessler** [comment]
A [right circular cone](https://en.wikipedia.org/wiki/Cone#Right_circular_cone) is easy and I'd be glad to add that myself so that you can focus on the harder work at MRIReco.jl, unless you want to make the PR yourself.

An oblique cone would be harder  because we'd need another parameter in its `Shape` type to describe how oblique.  I started to work on obliqueness for a [triangle(https://github.com/JuliaImageRecon/ImagePhantoms.jl/blob/74aba2a25b6e3205a85f61e4510af4bc4e2690f2/src/triangle.jl#L23)
 but abandoned it.
Ideally we'd have methods for the Radon transform and spectrum of the cone, but I don't know those functions.  (Do you?)  It's ok for now to have throwing placeholder methods that could be filled in later if someone needs them.

**Comment by tknopp** [comment]
> Ideally we'd have methods for the Radon transform and spectrum of the cone, but I don't know those functions. (Do you?) It's ok for now to have throwing placeholder methods that could be filled in later if someone needs them.

No, I don't have analytical methods for the  Radon transform and spectrum. But while it is great to have these for certain shapes I would also say that this should not be a show stopper for the introduction of other shapes.

**Comment by JeffFessler** [comment]
Is a right circular cone sufficient for you?  (Actually it doesn't have to be circular - we'll get ellipses for free, but is *right* sufficient?)

**Comment by tknopp** [comment]
yes for my application a right circular cone is sufficient. I would think we don't need to make the shape collection exhaustive but rather collect common shapes used in tomography applications so far.

**Comment by JeffFessler** [comment]
Closed by #62
